### ISSUE TITLE
Import script for new ingredients + type + filtering

### ISSUE THREAD

**Issue by rubikscuber** [flauschtrud/broccoli#196]
As a user (not programmer) I want to be able to import a list of ingredients based on a simple csv file - e.g. [ingredients.csv](https://github.com/rubikscuber/broccoli/blob/ingredient-list/ingredients.csv). This could enable non programmers to contribute to the ingredient list, translations, synonyms as well as seasonal calendar. 

The type could be used for filtering / as a category or simply color/group/style the ingredient names in the app.
Basically I just strive for a single file to edit to generate the resource files needed. Given a good structure, it could support the different languages directly.

This needs some further discussion and planning - what are your thoughts?

**Comment by flauschtrud** [comment]
I shortly thought about making the seasonal calendar editable but I chose not to for several reasons. First most users don't want to manage stuff like this themselves. It's also quite probable that they wouldn't even understand this feature (you would be surprised, I did some things that I thought were quite obvious and simple and my test users didn't get it and even thought there was a bug). An ingredient management system also comes with additional challenges, like backing up the custom data and merging it with the master data.

I also think that "simple CSV that generates all resources" is a contradiction. For example one dimension that is still missing in your CSV is the region, because there is not only one seasonal calendar, which would make it even more complex. That's why I prefer more modular solutions.

How would this approach make it easier to contribute? Unless there is some editable master CSV online a user couldn't share their data with others. But something like this would have to be moderated so we're basically back to tools like GitHub or Crowdin ;)

What exactly are the use cases that you want to accomplish with this additional data? Can you describe these with more detail (best with examples based on your CSV)?

I think a custom CSV file is both a very technical and highly individual approach and it's likely that a lot of use cases can be accomplished with things like hashtags (which are technically super simple but extremely individual). For example I tag my recipes with tags like "#pasta", "#party", "#christmas", "#vleisch" (veggie meat), and so on. So I can do searches like
- "#pasta #party" if I want to find any kind of pasta recipe that is suitable for many people
- "#vleisch broccoli" if I want to find a recipe that uses some kind of veggie meat and broccoli
- "#christmas cinnamon" for holiday recipes that use cinnamon

When I created the app a minimal design with still powerful features was very important to me, since I was missing that in other apps. I think there's still a lot of potential, especially regarding teaching users what they can actually do with the app (that's a tough one and I'm no UI/UX expert) or a better visual representation of tags for example. But I want to carry on with the minimal design philosophy.

**Comment by rubikscuber** [comment]
I tried the csv approach, because I did not really know in which files I would need to insert a new ingredient. To avoid having to edit multiple files + their translation equivalent I started with this. I agree that the CSV can get very technical and complicated.

I thought about crowdin as well, however, the extra registration stopped me do do this very quickly. + I think there was only polish available at the moment (please correct me if I am wrong).

Maybe a guideline / best practice description on how to add ingredients is enough for me. I had the other feature in mind when creating the CSV, so if this will not be implemented, then I would also not need to add many ingredients.

I like the design. Also the example for importing a recipe is very good. Though, I almost uninstalled the app, because I did not see this option at first. I'm glad I did find it in the end :)

**Comment by flauschtrud** [comment]
The translation with Crowdin went actually quite smoothly. I added a GitHub integration and Crowdin pulls updates to the English default texts and pushes the translations back to a translation branch automatically. I didn't add any more languages at the moment because there a no translators yet, but there's now some info in the readme file.

In case by add ingredients you mean to add seasonal foods to the seasonal calendar, I can add a how-to to the readme as well.

And like I said in the other post: ideas for improving the onboarding process of new users are very welcome here #197 I know the importing feature is not that easy to get at first.
